zingolib/src/lightclient.rs

```diff
@@ -271,10 +279,75 @@ impl LightClient {
 
         Self::create_with_new_wallet(config, latest_block)
     }
+
+    //        SAVE METHODS
+
+    async fn save_internal_rust(&self) -> Result<bool, ZingoLibError> {
+        self.save_internal_buffer().await?;
+        self.rust_write_save_buffer_to_file().await
+    }
+
+    async fn save_internal_buffer(&self) -> Result<(), ZingoLibError> {
+        let mut buffer: Vec<u8> = vec![];
```

Comment:
this line is actually kinda weird. i dont see why we dont just write buffer  = self.wallet.write... but i cant tell right now